test/events.js

```diff
@@ -22,6 +22,42 @@ describe('events', function () {
     })
   })
 
+  it.only('waits on connect until finished', function (done) {
+    var pool = new Pool()
+    var emittedClient = false
+    pool.on('connect', function (client) {
+      return new Promise(resolve => {
+        setTimeout(resolve, 50)
+      })
+      .then(() => {
+        emittedClient = client
+      })
```

Comment:
indentation still bugs me 🤓 
also, looking at current version you can also use
```
return Promise.delay(50)
    .then(() => emittedClient = client)
```

Reply:
Dont want to use bluebird promises, im gonna keep it simple as is..